### PR TITLE
Feat/tutorials

### PR DESCRIPTION
**Pull request by rosecers**
This is a draft PR to start merging tutorials framework into master branch

Rationale and detailed description of the changes:

- Added splashpage within documentation
- Wrote Learning tutorial
- Outlined tentative tutorials and tutorial maps

Tasks before review:

- [X] Add tests, examples, and complete documentation of any added functionality
    - [ ] Make sure the autogenerated documentation appears in Sphinx and is
          formatted correctly (ask @max-veit if you need help with this task).
    - [ ] Write additional documentation in the Sphinx (not docstrings!) to
          explain the feature and its usage in plain English
    - [X] Make sure the examples run (!) and produce the expected output
    - [ ] For bugfix pull requests, list here which tests have been added or re-enabled
          to verify the fix and catch future regressions as well as similar bugs
          elsewhere in the code.
- [X] Run `make lint` on the project, ensure it passes
    - [X] Run `make pretty-cpp` and `make pretty-python`, check that the
          auto-formatting is sensible
    - [X] Review variable names, make sure they're descriptive (ask a friend)
    - [X] Review all TODOs, convert to issues wherever possible
    - [ ] Make sure draft, in-progress, and commented-out code is moved to its
          own branch
- [X] If committing any code in Jupyter/IPython notebooks, install and run `nbstripout`
- [X] Merge with master and resolve any conflicts
- [X] (anything else that's still in progress)

Remaining tasks, out of scope of this pull request: (optional)

- Continue writing and receiving feedback on tutorials

### CODE REVIEW

**Comment by Luthaf** [comment]
Could you rebase this branch on top of master, there are multiple commits in this PR that have already been merged, and this would help fixing the conflicts

**Comment by mastricker** [comment]
Starting to comment.

`make doc` results in 
```
/home/markus/repos/review/librascal/docs/source/tutorials/SOAP_CSD500.rst: WARNING: document not readable. Ignored.
/home/markus/repos/review/librascal/docs/source/tutorials/SOAP_CSD500.rst: WARNING: document not readable. Ignored.
/home/markus/repos/review/librascal/docs/source/SOAP.rst:362: WARNING: Footnote [7] is not referenced.
/home/markus/repos/review/librascal/docs/source/SOAP.rst:355: WARNING: Unknown target name: "8".

Exception occurred:
  File "/home/markus/.local/lib/python3.7/site-packages/breathe/renderer/sphinxrenderer.py", line 47, in parse_definition
    ast = parser.parse_declaration("class", "class")
TypeError: parse_declaration() takes 2 positional arguments but 3 were given
```
Not sure how to handle that one.

Further:
```
/home/markus/repos/review/librascal/docs/source/tutorials/SOAP_CSD500.rst: WARNING: document not readable. Ignored.
```
I'll keep updating this.

**Comment by mastricker** [comment]
Just because I am currently looking at it: The last push with the notebook was not stripped. There is calculated data inside as well as figures "Updating PPT".

**Comment by mastricker** [comment]
Above issues have been solved, but can not compile documentation with the following error:
```
from rascal.representations import SphericalInvariants as SOAP
------------------

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
~/repos/review/librascal/docs/source/tutorials/utilities/general_utils.py in <module>
     17 try:
---> 18     from rascal.representations import SphericalInvariants as SOAP
     19 except:

ModuleNotFoundError: No module named 'rascal'

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-561b50b9a161> in <module>
     17 wd=get_ipython().getoutput('(pwd)')
     18 sys.path.append(f'{wd[0]}/utilities')
---> 19 from general_utils import *
     20 from rascal.representations import SphericalInvariants as SOAP

~/repos/review/librascal/docs/source/tutorials/utilities/general_utils.py in <module>
     18     from rascal.representations import SphericalInvariants as SOAP
     19 except:
---> 20     from rascal.representations import SOAP
     21 
     22 # These are some imports/helpers for all of the tutorials

ModuleNotFoundError: No module named 'rascal'
ModuleNotFoundError: No module named 'rascal'
```

Did you make a system-wide install of the library?

**Comment by rosecers** [comment]
> Just because I am currently looking at it: The last push with the notebook was not stripped. There is calculated data inside as well as figures "Updating PPT".

That was unintentional, but does bring up a point. For the notebooks in the documentation, unless they are run, the documents will not reflect the outputs or cause a timeout if the operation takes too long. So unless we want the users to go through and run the notebooks before building docs, we'll need them pre-run before make. Thoughts?

As for SOAP_CSD500.rst, that will be removed.

**Comment by rosecers** [comment]
> Above issues have been solved, but can not compile documentation with the following error:
> 
> ```
> from rascal.representations import SphericalInvariants as SOAP
> ------------------
> 
> ---------------------------------------------------------------------------
> ModuleNotFoundError                       Traceback (most recent call last)
> ~/repos/review/librascal/docs/source/tutorials/utilities/general_utils.py in <module>
>      17 try:
> ---> 18     from rascal.representations import SphericalInvariants as SOAP
>      19 except:
> 
> ModuleNotFoundError: No module named 'rascal'
> 
> During handling of the above exception, another exception occurred:
> 
> ModuleNotFoundError                       Traceback (most recent call last)
> <ipython-input-1-561b50b9a161> in <module>
>      17 wd=get_ipython().getoutput('(pwd)')
>      18 sys.path.append(f'{wd[0]}/utilities')
> ---> 19 from general_utils import *
>      20 from rascal.representations import SphericalInvariants as SOAP
> 
> ~/repos/review/librascal/docs/source/tutorials/utilities/general_utils.py in <module>
>      18     from rascal.representations import SphericalInvariants as SOAP
>      19 except:
> ---> 20     from rascal.representations import SOAP
>      21 
>      22 # These are some imports/helpers for all of the tutorials
> 
> ModuleNotFoundError: No module named 'rascal'
> ModuleNotFoundError: No module named 'rascal'
> ```
> 
> Did you make a system-wide install of the library?

Yes, I did. I would assume this is what we want users to do?

**Comment by Luthaf** [comment]
> Yes, I did. I would assume this is what we want users to do?

I agree with @rosecers that using a 'global' install is better in the docs/tutorials/examples. 

Fixing the above error it is just a matter of fixing the documentation build by adding the right directory to `sys.path` in `docs/sources/conf.py.in`

**Comment by mastricker** [comment]
I have no strong opinion on what we settle at the end, but we had the discussion about that (before you arrived). And as far as I can recall we settled on that you _can_ install it system wide and we provide the option, but it should not be _necessary_ to do so for examples, tutorials, tests, etc.

@Felix for example went through a lot of pain with the python binding tests etc. to make them work by referring to the build folder and adding it to the system path `sys.path.insert(0,...)` something like that, so that it is self contained without the `make install` - for which I need to provide root rights and I am not sure I want to do that to be able to look at a tutorial of a library.

While most of the people do not belong into that category, some don't trust other people's code with a system wide install.

So I do disagree on that and our past decisions (before you joined) also. I'll put it on the list for Wednesday so we can discuss in person. It is probably easier.

**Comment by Luthaf** [comment]
I put global inside quotes because this do not need to be an actual system-wide install to work. 

For example, #165 sets PYTHONPATH for tests to ensure they find the right version of rascal. The same could be done for the user-facing documentation. But this is diverging from the PR, so let's discuss it in person tomorrow.

**Comment by Luthaf** [comment]
It looks like this PR includes files that are already on master (for example `docs/source/Doxyfile` is now in `docs/Doxyfile.in`, etc.
